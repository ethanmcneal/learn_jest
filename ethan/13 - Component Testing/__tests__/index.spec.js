import React from 'react';
import List from '../index';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

// NO EDITS NECESSARY IN THIS FILE

describe('<List />', () => {
  it('should render an `li` for each data item', () => {
    const data = [
      { key: 1, name: 'Item 1', selected: false },
      { key: 2, name: 'Item 2', selected: false },
      { key: 3, name: 'Item 3', selected: false },
      { key: 4, name: 'Item 4', selected: false },
    ];
    const wrapper = shallow(<List data={data} onClick={jest.fn()} />);
    expect(wrapper.find('li').length).toEqual(data.length);
  });

  it('should call `onClick` prop with item key on item clicked', () => {
    const data = [
      { key: 1, name: 'Item 1', selected: false },
      { key: 2, name: 'Item 2', selected: false },
      { key: 3, name: 'Item 3', selected: true },
      { key: 4, name: 'Item 4', selected: false },
    ];
    const mockClickHandler = jest.fn();
    const wrapper = shallow(<List data={data} onClick={mockClickHandler} />);
    wrapper.find('li.selected').simulate('click');
    expect(mockClickHandler).lastCalledWith(3);
  });

  it('should render consistently', () => {
    let data = [
      { key: 1, name: 'Item 1', selected: false },
      { key: 2, name: 'Item 2', selected: true },
      { key: 3, name: 'Item 3', selected: false },
      { key: 4, name: 'Item 4', selected: false },
    ];
    let component = renderer.create(<List data={data} onClick={jest.fn()} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    data = [
      { key: 1, name: 'Item 1', selected: false },
      { key: 3, name: 'Item 3', selected: false },
      { key: 4, name: 'Item 4', selected: true },
      { key: 5, name: 'Item 5', selected: false },
    ];
    component = renderer.create(<List data={data} onClick={jest.fn()} />);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
