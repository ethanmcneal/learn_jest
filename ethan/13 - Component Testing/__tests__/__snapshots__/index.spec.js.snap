exports[`<List /> should render consistently 1`] = `
<ul>
  <li
    className=""
    onClick={[Function]}>
    Item 1
  </li>
  <li
    className="selected"
    onClick={[Function]}>
    Item 2
  </li>
  <li
    className=""
    onClick={[Function]}>
    Item 3
  </li>
  <li
    className=""
    onClick={[Function]}>
    Item 4
  </li>
</ul>
`;

exports[`<List /> should render consistently 2`] = `
<ul>
  <li
    className=""
    onClick={[Function]}>
    Item 1
  </li>
  <li
    className=""
    onClick={[Function]}>
    Item 3
  </li>
  <li
    className="selected"
    onClick={[Function]}>
    Item 4
  </li>
  <li
    className=""
    onClick={[Function]}>
    Item 5
  </li>
</ul>
`;
